File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        int isTrue=0;
        for(int i=0; i<s.length(); i++){
            isTrue += s.charAt(i)=='('? 1: -1;
            if(isTrue<0){
                return false;
            }
        }
        return isTrue==0? true: false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack=[]
    for i in s:
        if i==')' and  (len(stack)!=0):
            stack.pop()
        elif i=='(':
            stack.append(i)
        else:
            return False
    if len(stack)==0:
        return True
    return False